platforms/ide/problems-api/src/main/java/org/gradle/internal/problems/failure/StackTraceRelevance.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.problems.failure;
+
+public enum StackTraceRelevance {
+    USER_CODE,
+    RUNTIME,
+    SYSTEM
+}
```

Comment:
I am open to better naming here

Reply:
What's the difference between "runtime" and "system"? What would a consumer do with this information?

Reply:
The simplisitic idea is that the "system" things we simply drop from the displayed stacktraces, and the "runtime" things are still useful to preserve coherency but with lower priority than user code.

Reply:
I'm not sure I'd bother with that distinction, at least until we get feedback that something like "user code" and "internal code" is not sufficient.

Reply:
The distinction between "user code" and "internal code" currently is a feature supported only by the CC report, because we control the UI. 

If we do not introduce the "system" category to throw away parts of the stack as we did before, then those stack frames will appear in the console, IDE, and Build Scan output. It does not happen in this PR, but it would happen in the follow-ups because the failure factory will be reused more widely for the original stacktrace-handling consolidation purposes.

Reply:
I would merge "runtime" and "system" into a single "internal" category. The stack frames would not appear in the console, or the IDE, or build scan output, or the CC report (unless the user expands them of course).

I'm not seeing the benefit of two separate internal categories. I'm not sure why I would care as a user of this information.